### PR TITLE
Add notes on acronyms and initialisms to naming

### PR DESCRIPTION
**Pull request by kesne**
This was previously ambiguous. Let's reduce decision fatigue and just pick one.

### CODE REVIEW

**Comment by goatslacker** [comment]
👍

File: README.md [hunk 1]
```diff
@@ -2573,6 +2573,29 @@ Other Style Guides
...
+    // bad
+    import SmsContainer from './containers/SmsContainer';
+    
+    // bad 
+    const HttpRequests = [
+      // ...
+    ];
+    
+    // good
+    import SMSContainer from './containers/SMSContainer';
```
Comment (ljharb) [inline README.md]: i wonder if for both of these, we should include a "best" example as well that renames these things so as not to run into this question?

Comment (kesne) [inline README.md]: Maybe we can do that for double-acronyms? i.e. `XMLHTTPRequest` renamed to `HTTPRequest`?

Comment (ljharb) [inline README.md]: I'm thinking that for example, "TextMessageContainer" might be better than "SMSContainer", or "Request" better than "HTTPRequest" (but with better names).

Comment (kesne) [inline README.md]: Sounds good. I'll also add a note about allowing all-lowercase to support non-constructors.

Comment (mvolk) [inline README.md]: If we're going with uppercase acronyms, perhaps we should also provide guidance regarding running multiple acronyms together in a name?

Comment (ljharb) [inline README.md]: Definitely - my preferred guidance is "pick a name where that's not a problem" :-p

**Pull request merged**